azurelinuxagent/common/protocol/wire.py

```diff
@@ -837,70 +828,59 @@ def save_goal_state(incarnation, xml_text):
 
         raise ProtocolError("Exceeded max retry updating goal state")
 
-    def get_goal_state(self):
-        if self.goal_state is None:
-            incarnation_file = os.path.join(conf.get_lib_dir(),
-                                            INCARNATION_FILE_NAME)
-            incarnation = self.fetch_cache(incarnation_file)
+    def set_goal_state(self, new_goal_state):
+        if new_goal_state is not None:
```

Comment:
why do we need the check for None? thanks

Reply:
you should probably raise; if for any reason one of those values is None then the goal state would be inconsistent: would have the previous values

Reply:
Good point, I'll modify the code